### PR TITLE
Fixing "edit on github" links for stable version builds.

### LINKED ISSUE #1062: "edit on github" not working for the 'stable' branch
Hi.
The "edit on github" link does not work for the stable documentation version, as it tries to open the non- existent `stable` branch on github.

As I understand it, both  `latest` and `stable` are references, latest to master, and stable to the latest tagged version ( see https://docs.readthedocs.org/en/latest/versions.html ). However it seems that only `latest` has special logic to deal with the edit on github feature
https://github.com/rtfd/readthedocs.org/blob/e67465175d981d19bdd98e911d02c85e7009edff/readthedocs/builds/utils.py#L40
So the solution would be to apply the same logic, and have `stable` point to the latest tagged version.

### PR DESCRIPTION
**Pull request by gregmuellegger**
That fixes #1062.

### CODE REVIEW

File: readthedocs/builds/utils.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def get_bitbucket_username_repo(version, repo_url=None):
 
 def get_vcs_version_slug(version):
     slug = None
-    if version.slug == LATEST:
+    if version.slug in NON_REPOSITORY_VERSIONS:
```
Comment (ericholscher) [inline readthedocs/builds/utils.py]: Not sure if this logic will work. The logic that is after here only applies to 'latest', and will just end up with stable linking to the save version as latest.

Stable is a different branch/tag than latest, and should be treated differently.

**Comment by ericholscher** [comment]
I believe the proper way to fix this is to pass along either the computed tag, or link to the commit on GitHub.

In general, having a tag means that it isn't really even "editable" on GitHub anymore, so it's a bit of a misnomer. Branches will always be up to date and editable, however with a tag, I'm not sure what the best solution is there. It's likely the content will have moved on, so you can see what it was at a certain point in time, but the verbiage is likely incorrect.

**Comment by gregmuellegger** [comment]
Oh, ofcourse! It was a short shot on this :)
I think we could link to the direct commit if it's a tag or the `stable` identifier.

GitHub will show this when viewing a specific commit:
screen2

At least if one knows about git and how it works you can figure it out yourself that you need to look up the file in the master branch.

We could also change the wording to be "View source on GitHub" to make things clearer. However "source" could imply that it links to the source code of the API documented on the current page.

**Comment by ericholscher** [comment]
Yea, it's a tricky UI/UX problem, but I think pointing at the commit is the most reasonable thing.

**Comment by gregmuellegger** [comment]
This includes now a special handling for stable branches, and returns the commit hash for stable instead of the branch name. We could do better and return the tag name here, but I think we do not track the tag name for stable branches so far in a good way.

I also added a few refactoring commits that move some util functions that only operate on the version model and nothing else to methods of `Version`.

**IMO this PR is ready for merge. Anyone up for a final review?**

After this PR is merged and in production we should create a PR for the rtd theme that uses the ``github/bitbucket_version_is_editable`` flag to display an alternative wording.

I think in future development we should add a new field `commit_name` to the `Version` model to track exactly that. `identifier` should be the hash, and `commit_name` the appropriate tag/branch name if it exists, whereas `slug` should only be used in the readthedocs urls. But that is scope of another refactor.

**Comment by ericholscher** [comment]
:+1: on the PR, and the idea of seperating commit & id in the version.

**Pull request merged**